### PR TITLE
8.get/api/reviews

### PR DESCRIPTION
**Pull request by DeanH1990**
This PR adds getReviews function, the default sort is by review age, it allows sorting by category. It adds a comment count to each review. It responds with an error if a invalid order query is used. The tests test each of these things.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -77,16 +77,23 @@ describe('app', () => {
                         expect(reviews).toBeSortedBy('created_at', { descending: true })
                     })
                 })
-                test.only('status 200: allows reviews to be sorted by category, asc by default', () => {
+                test('status 200: allows reviews to be sorted by category, asc by default', () => {
```
Comment (B7Bprog) [inline __tests__/app.test.js]: Sorting by category would mean you want results from a certain category. For example: '?category=social deduction'

**Review by B7Bprog — COMMENTED** [review]
Please check my comment, then write a test (200) with an existing category, then one with status code 404 for a non-existent category, and lastly one with status code 404 when category exists. but there are no relevant reviews.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -42,6 +42,70 @@ describe('app', () => {
...
+                })
+                test('status 404: responds with error if passed an invalid category query', () => {
+                    return request(app)
+                    .get('/api/reviews?category=bananas')
+                    .expect(404)
+                    .then(({ body }) => {
+                        expect(body.msg).toBe('Not found')
+                    })
+                })
+                test('status 404: responds with error if valid category but no results', () => {
```
Comment (corkesw) [inline __tests__/app.test.js]: You should change this assertion, if the category exists then I would expect a 200 status and an empty array to be returned

File: models/app.models.js [hunk 3]
```diff
@@ -43,4 +43,35 @@ exports.patchReviewVotesById = (review_id, inc_votes) => {
         }
         return review
     })
+}
+
+exports.selectReviews = (category) => {
+    const allowedCategory = ['euro game', 'social deduction', 'dexterity', "children's games"];
```
Comment (corkesw) [inline models/app.models.js]: Hard coding the categories will cause problems if we add more categories later - you should get this information from the database

File: models/app.models.js [hunk 4]
```diff
@@ -43,4 +43,35 @@ exports.patchReviewVotesById = (review_id, inc_votes) => {
...
+    const queryValues = [];
+
+    if (category) {
+        queryStr += ` WHERE reviews.category = $1`
+        queryValues.push(category)
+    }
+
+    queryStr += ` GROUP BY reviews.review_id ORDER BY created_at DESC;`
+
+    return db.query(queryStr, queryValues).then(({ rows: reviews }) => {
```
Comment (corkesw) [inline models/app.models.js]: This logic will cause the suggested test to fail

**Review by corkesw — COMMENTED** [review]
Great work again here! There is another test to add and I would avoid hard coding of categories. In general terms though, this code is of a high standard and is easy to read throughout.

**Pull request merged**